ngraph/core/src/graph_util.cpp

```diff
@@ -405,6 +406,32 @@ std::shared_ptr<ngraph::Function> ngraph::clone_function(const ngraph::Function&
     // clone function operations
     clone_nodes(func.get_ops(), node_map);
 
+    // clone variables
+    auto variables = func.get_variables();
+    VariableVector cloned_vars;
+    std::map<std::string, std::shared_ptr<Variable>> var_map;
+    for (const auto& var : variables)
+    {
+        auto cloned_var = std::make_shared<Variable>(
```

Comment:
can we use clone_with_new_inputs to avoid manual node creation?

Reply:
Variable class is not inherited from Node and doesn't have clone_with_new_inputs method.
and we also can't use clone_with_new_inputs methods of Assign/ReadValue ops because Variable belongs to ngraph::function